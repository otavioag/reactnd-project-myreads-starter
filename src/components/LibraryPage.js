import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Spin } from 'antd';
import BookShelf from './book/BookShelf';

const LibraryPage = (props) => (
  <div className='list-books'>
    <div className='list-books-title'>
      <h1>MyReads</h1>
    </div>
    <Spin spinning={props.loading} size='large' tip='Loading...' >
      <div className='list-books-content'>
        <BookShelf
          title='Currently Reading'
          shelf='currentlyReading'
          books={props.booksInShelf}
          handleShelfChange={props.handleShelfChange}
        />
        <BookShelf
          title='Want to Read'
          shelf='wantToRead'
          books={props.booksInShelf}
          handleShelfChange={props.handleShelfChange}
        />
        <BookShelf
          title='Read'
          shelf='read'
          books={props.booksInShelf}
          handleShelfChange={props.handleShelfChange}
        />
      </div>
    </Spin>
    <div className='open-search'>
      <Link to='/search'
        className='search-book' >
        <button>Add a book</button>
      </Link>
    </div>
  </div>
);

LibraryPage.propTypes = {
  booksInShelf: PropTypes.array.isRequired,
  handleShelfChange: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};

export default LibraryPage;