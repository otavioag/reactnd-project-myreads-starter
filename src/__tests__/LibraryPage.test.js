import React from 'react';
import { shallow } from 'enzyme';
import LibraryPage from '../components/LibraryPage';


describe('LibraryPage', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<LibraryPage
      booksInShelf={[]}
      handleShelfChange={jest.fn()}
      loading={true}/>
    );
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});