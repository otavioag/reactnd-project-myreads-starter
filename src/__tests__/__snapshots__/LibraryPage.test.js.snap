// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryPage renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LibraryPage
    booksInShelf={Array []}
    handleShelfChange={[MockFunction]}
    loading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="list-books-title"
        >
          <h1>
            MyReads
          </h1>
        </div>,
        <Spin
          size="large"
          spinning={true}
          tip="Loading..."
          wrapperClassName=""
        >
          <div
            className="list-books-content"
          >
            <BookShelf
              books={Array []}
              handleShelfChange={[MockFunction]}
              shelf="currentlyReading"
              title="Currently Reading"
            />
            <BookShelf
              books={Array []}
              handleShelfChange={[MockFunction]}
              shelf="wantToRead"
              title="Want to Read"
            />
            <BookShelf
              books={Array []}
              handleShelfChange={[MockFunction]}
              shelf="read"
              title="Read"
            />
          </div>
        </Spin>,
        <div
          className="open-search"
        >
          <Link
            className="search-book"
            replace={false}
            to="/search"
          >
            <button>
              Add a book
            </button>
          </Link>
        </div>,
      ],
      "className": "list-books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            MyReads
          </h1>,
          "className": "list-books-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MyReads",
          },
          "ref": null,
          "rendered": "MyReads",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="list-books-content"
          >
            <BookShelf
              books={Array []}
              handleShelfChange={[MockFunction]}
              shelf="currentlyReading"
              title="Currently Reading"
            />
            <BookShelf
              books={Array []}
              handleShelfChange={[MockFunction]}
              shelf="wantToRead"
              title="Want to Read"
            />
            <BookShelf
              books={Array []}
              handleShelfChange={[MockFunction]}
              shelf="read"
              title="Read"
            />
          </div>,
          "size": "large",
          "spinning": true,
          "tip": "Loading...",
          "wrapperClassName": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="currentlyReading"
                title="Currently Reading"
              />,
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="wantToRead"
                title="Want to Read"
              />,
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="read"
                title="Read"
              />,
            ],
            "className": "list-books-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "books": Array [],
                "handleShelfChange": [MockFunction],
                "shelf": "currentlyReading",
                "title": "Currently Reading",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "books": Array [],
                "handleShelfChange": [MockFunction],
                "shelf": "wantToRead",
                "title": "Want to Read",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "books": Array [],
                "handleShelfChange": [MockFunction],
                "shelf": "read",
                "title": "Read",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="search-book"
            replace={false}
            to="/search"
          >
            <button>
              Add a book
            </button>
          </Link>,
          "className": "open-search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Add a book
            </button>,
            "className": "search-book",
            "replace": false,
            "to": "/search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add a book",
            },
            "ref": null,
            "rendered": "Add a book",
            "type": "button",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="list-books-title"
          >
            <h1>
              MyReads
            </h1>
          </div>,
          <Spin
            size="large"
            spinning={true}
            tip="Loading..."
            wrapperClassName=""
          >
            <div
              className="list-books-content"
            >
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="currentlyReading"
                title="Currently Reading"
              />
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="wantToRead"
                title="Want to Read"
              />
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="read"
                title="Read"
              />
            </div>
          </Spin>,
          <div
            className="open-search"
          >
            <Link
              className="search-book"
              replace={false}
              to="/search"
            >
              <button>
                Add a book
              </button>
            </Link>
          </div>,
        ],
        "className": "list-books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              MyReads
            </h1>,
            "className": "list-books-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MyReads",
            },
            "ref": null,
            "rendered": "MyReads",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="list-books-content"
            >
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="currentlyReading"
                title="Currently Reading"
              />
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="wantToRead"
                title="Want to Read"
              />
              <BookShelf
                books={Array []}
                handleShelfChange={[MockFunction]}
                shelf="read"
                title="Read"
              />
            </div>,
            "size": "large",
            "spinning": true,
            "tip": "Loading...",
            "wrapperClassName": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <BookShelf
                  books={Array []}
                  handleShelfChange={[MockFunction]}
                  shelf="currentlyReading"
                  title="Currently Reading"
                />,
                <BookShelf
                  books={Array []}
                  handleShelfChange={[MockFunction]}
                  shelf="wantToRead"
                  title="Want to Read"
                />,
                <BookShelf
                  books={Array []}
                  handleShelfChange={[MockFunction]}
                  shelf="read"
                  title="Read"
                />,
              ],
              "className": "list-books-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "books": Array [],
                  "handleShelfChange": [MockFunction],
                  "shelf": "currentlyReading",
                  "title": "Currently Reading",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "books": Array [],
                  "handleShelfChange": [MockFunction],
                  "shelf": "wantToRead",
                  "title": "Want to Read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "books": Array [],
                  "handleShelfChange": [MockFunction],
                  "shelf": "read",
                  "title": "Read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="search-book"
              replace={false}
              to="/search"
            >
              <button>
                Add a book
              </button>
            </Link>,
            "className": "open-search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button>
                Add a book
              </button>,
              "className": "search-book",
              "replace": false,
              "to": "/search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add a book",
              },
              "ref": null,
              "rendered": "Add a book",
              "type": "button",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
